### ISSUE TITLE
Error with PackageInfoVersioner

### ISSUE THREAD

**Issue by apizz** [facebook/Recipes-for-AutoPkg#47]
To be clear, up until very recently I was still on autopkg 1.4.1, so all of my recipe testing was done pre-2.0.  Prior to upgrading to the latest (2.1), I created a recipe for importing Pro Tools into munki.  This processes not only the Pro Tools PKG installer but also 2 other PKGs in the DMG which are version specific to the included version of Pro Tools. As such the recipe grabs the PKG versions of these two other installers so they can be included with the Pro Tools pkginfo file.  After upgrading to 2.1, I now get an error.

System info:
autopkg: 2.1
macOS: 10.14.6

Applicable lines in the recipe: https://github.com/autopkg/apizz-recipes/blob/master/ProTools/ProToolsImport.munki.recipe#L64-L139

I was reviewing my recipe list run and noted that the run failed to continue after running PackageInfoVersioner the second time it's called in the recipe. Given the error, my assumption is it has to do with the processor rather than autopkg, but I could be wrong given my lack of Python know-how. If it turns out to be an autopkg thing, I can move this issue there.

Interestingly, PackageInfoVersioner does not fail on the first Codecs pkg, but fails on the second HD driver package. Not sure what info would be helpful for isolating what's going on or what's different with this second package. I don't expect you to have the software, so if want a copy of the pkg let me know.

```
autopkg run ProToolsImport.or.munki -vv
Processing ProToolsImport.or.munki...
FileFinder
{'Input': {'pattern': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/*.dmg'}}
FileFinder: No value supplied for find_method, setting default value of: glob
FileFinder: Found file match: '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg' from globbed '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/*.dmg'
{'Output': {'found_filename': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg'}}
CodeSignatureVerifier
{'Input': {'expected_authority_names': ['Developer ID Installer: Avid '
                                        'Technology Inc (4UYUA773XD)',
                                        'Developer ID Certification Authority',
                                        'Apple Root CA'],
           'input_path': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Install*.pkg'}}
CodeSignatureVerifier: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
CodeSignatureVerifier: Using path '/private/tmp/dmg.qgNL5D/Install Pro Tools 2019.12.0.pkg' matched from globbed '/private/tmp/dmg.qgNL5D/Install*.pkg'.
CodeSignatureVerifier: Verifying installer package signature...
CodeSignatureVerifier: Package "Install Pro Tools 2019.12.0.pkg":
CodeSignatureVerifier:    Status: signed by a certificate trusted by Mac OS X
CodeSignatureVerifier:    Certificate Chain:
CodeSignatureVerifier:     1. Developer ID Installer: Avid Technology Inc (4UYUA773XD)
CodeSignatureVerifier:        SHA1 fingerprint: 52 C7 74 17 B8 F7 11 F5 92 B1 B3 F3 9A C4 A5 54 E8 B1 86 36
CodeSignatureVerifier:        -----------------------------------------------------------------------------
CodeSignatureVerifier:     2. Developer ID Certification Authority
CodeSignatureVerifier:        SHA1 fingerprint: 3B 16 6C 3B 7D C4 B7 51 C9 FE 2A FA B9 13 56 41 E3 88 E1 86
CodeSignatureVerifier:        -----------------------------------------------------------------------------
CodeSignatureVerifier:     3. Apple Root CA
CodeSignatureVerifier:        SHA1 fingerprint: 61 1E 5B 66 2C 59 3A 08 FF 58 D1 4A E2 24 52 D1 98 DF 6C 60
CodeSignatureVerifier: 
CodeSignatureVerifier: Signature is valid
CodeSignatureVerifier: Authority name chain is valid
{'Output': {}}
com.github.homebysix.VersionSplitter/VersionSplitter
{'Input': {'index': -2,
           'split_on': '_',
           'version': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg'}}
VersionSplitter: Split version: 2019.12.0
{'Output': {'version': '2019.12.0'}}
Copier
{'Input': {'destination_path': '/Volumes/EXT_DATA/Autopkgr/Cache/local.munki.or.ProToolsImport/ProTools2019-2019.12.0.pkg',
           'overwrite': True,
           'source_path': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Install*.pkg'}}
Copier: Parsed dmg results: dmg_path: /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg, dmg: .dmg/, dmg_source_path: Install*.pkg
Copier: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
Copier: Using path '/private/tmp/dmg.an3QU2/Install Pro Tools 2019.12.0.pkg' matched from globbed '/private/tmp/dmg.an3QU2/Install*.pkg'.
Copier: Copied /private/tmp/dmg.an3QU2/Install Pro Tools 2019.12.0.pkg to /Volumes/EXT_DATA/Autopkgr/Cache/local.munki.or.ProToolsImport/ProTools2019-2019.12.0.pkg
{'Output': {}}
FileFinder
{'Input': {'find_method': 'glob',
           'pattern': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Codec '
                      'Installers/*.pkg'}}
FileFinder: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
FileFinder: Found file match: '/private/tmp/dmg.vbKpCd/Codec Installers/Install Avid Codecs LE.pkg' from globbed '/private/tmp/dmg.vbKpCd/Codec Installers/*.pkg'
FileFinder: DMG-relative file match: 'Codec Installers/Install Avid Codecs LE.pkg'
{'Output': {'dmg_found_filename': 'Codec Installers/Install Avid Codecs LE.pkg',
            'found_filename': '/private/tmp/dmg.vbKpCd/Codec '
                              'Installers/Install Avid Codecs LE.pkg'}}
FlatPkgUnpacker
{'Input': {'destination_path': '/Volumes/EXT_DATA/Autopkgr/Cache/local.munki.or.ProToolsImport/unpack',
           'flat_pkg_path': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Codec '
                            'Installers/Install Avid Codecs LE.pkg'}}
FlatPkgUnpacker: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
FlatPkgUnpacker: Unpacked /private/tmp/dmg.KuH5PD/Codec Installers/Install Avid Codecs LE.pkg to /Volumes/EXT_DATA/Autopkgr/Cache/local.munki.or.ProToolsImport/unpack
{'Output': {}}
com.facebook.autopkg.shared/PackageInfoVersioner
{'Input': {'package_info_path': '/Volumes/EXT_DATA/Autopkgr/Cache/local.munki.or.ProToolsImport/unpack/AvidCodecsLE_component.pkg/PackageInfo'}}
PackageInfoVersioner: Found version b'2.7.3.39170'
{'Output': {'version': b'2.7.3.39170'}}
FileFinder
{'Input': {'find_method': 'glob',
           'pattern': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Driver '
                      'Installers/*.pkg'}}
FileFinder: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
FileFinder: Found file match: '/private/tmp/dmg.4iSMDD/Driver Installers/Install Avid HD Driver.pkg' from globbed '/private/tmp/dmg.4iSMDD/Driver Installers/*.pkg'
FileFinder: DMG-relative file match: 'Driver Installers/Install Avid HD Driver.pkg'
{'Output': {'dmg_found_filename': 'Driver Installers/Install Avid HD '
                                  'Driver.pkg',
            'found_filename': '/private/tmp/dmg.4iSMDD/Driver '
                              'Installers/Install Avid HD Driver.pkg'}}
FlatPkgUnpacker
{'Input': {'destination_path': '/Volumes/EXT_DATA/Autopkgr/Cache/local.munki.or.ProToolsImport/unpack',
           'flat_pkg_path': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Driver '
                            'Installers/Install Avid HD Driver.pkg'}}
FlatPkgUnpacker: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
FlatPkgUnpacker: Unpacked /private/tmp/dmg.hj8yld/Driver Installers/Install Avid HD Driver.pkg to /Volumes/EXT_DATA/Autopkgr/Cache/local.munki.or.ProToolsImport/unpack
{'Output': {}}
com.facebook.autopkg.shared/PackageInfoVersioner
Traceback (most recent call last):
  File "/usr/local/bin/autopkg", line 2887, in <module>
    sys.exit(main(sys.argv))
  File "/usr/local/bin/autopkg", line 2881, in main
    exit(subcommands[verb]["function"](argv))
  File "/usr/local/bin/autopkg", line 2398, in run_recipes
    autopackager.process(recipe)
  File "/Library/AutoPkg/autopkglib/__init__.py", line 664, in process
    processor.inject(step.get("Arguments", {}))
  File "/Library/AutoPkg/autopkglib/__init__.py", line 465, in inject
    update_data(self.env, key, value)
  File "/Library/AutoPkg/autopkglib/__init__.py", line 375, in update_data
    a_dict[key] = do_variable_substitution(value)
  File "/Library/AutoPkg/autopkglib/__init__.py", line 357, in do_variable_substitution
    item = RE_KEYREF.sub(getdata, item)
TypeError: sequence item 0: expected str instance, bytes found
```

**Comment by apizz** [comment]
My suspicion is that because I call PkgInfoVersioner twice in the same recipe that the previously saved version configured as bytes is problematic when used again.

Tried supplying a `version = ''` to reset the version variable, but no luck.

**Comment by apizz** [comment]
Can confirm that when PkgInfoVersioner is run more than once without `version` being configured as bytes (removing `.encode("UTF-8"`) no error is produced.

**Comment by apizz** [comment]
Output after change:
```
autopkg run /Users/<user>/Downloads/ProToolsTest.munki.recipe -vv
Processing /Users/<user>/Downloads/ProToolsTest.munki.recipe...
WARNING: /Users/<user>/Downloads/ProToolsTest.munki.recipe is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...
FileFinder
{'Input': {'pattern': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/*.dmg'}}
FileFinder: No value supplied for find_method, setting default value of: glob
FileFinder: Found file match: '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg' from globbed '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/*.dmg'
{'Output': {'found_filename': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg'}}
CodeSignatureVerifier
{'Input': {'expected_authority_names': ['Developer ID Installer: Avid '
                                        'Technology Inc (4UYUA773XD)',
                                        'Developer ID Certification Authority',
                                        'Apple Root CA'],
           'input_path': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Install*.pkg'}}
CodeSignatureVerifier: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
CodeSignatureVerifier: Using path '/private/tmp/dmg.7MemfD/Install Pro Tools 2019.12.0.pkg' matched from globbed '/private/tmp/dmg.7MemfD/Install*.pkg'.
CodeSignatureVerifier: Verifying installer package signature...
CodeSignatureVerifier: Package "Install Pro Tools 2019.12.0.pkg":
CodeSignatureVerifier:    Status: signed by a certificate trusted by Mac OS X
CodeSignatureVerifier:    Certificate Chain:
CodeSignatureVerifier:     1. Developer ID Installer: Avid Technology Inc (4UYUA773XD)
CodeSignatureVerifier:        SHA1 fingerprint: 52 C7 74 17 B8 F7 11 F5 92 B1 B3 F3 9A C4 A5 54 E8 B1 86 36
CodeSignatureVerifier:        -----------------------------------------------------------------------------
CodeSignatureVerifier:     2. Developer ID Certification Authority
CodeSignatureVerifier:        SHA1 fingerprint: 3B 16 6C 3B 7D C4 B7 51 C9 FE 2A FA B9 13 56 41 E3 88 E1 86
CodeSignatureVerifier:        -----------------------------------------------------------------------------
CodeSignatureVerifier:     3. Apple Root CA
CodeSignatureVerifier:        SHA1 fingerprint: 61 1E 5B 66 2C 59 3A 08 FF 58 D1 4A E2 24 52 D1 98 DF 6C 60
CodeSignatureVerifier: 
CodeSignatureVerifier: Signature is valid
CodeSignatureVerifier: Authority name chain is valid
{'Output': {}}
com.github.homebysix.VersionSplitter/VersionSplitter
{'Input': {'index': -2,
           'split_on': '_',
           'version': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg'}}
VersionSplitter: Split version: 2019.12.0
{'Output': {'version': '2019.12.0'}}
Copier
{'Input': {'destination_path': '/Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/ProTools2019-2019.12.0.pkg',
           'overwrite': True,
           'source_path': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Install*.pkg'}}
Copier: Parsed dmg results: dmg_path: /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg, dmg: .dmg/, dmg_source_path: Install*.pkg
Copier: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
Copier: Using path '/private/tmp/dmg.Ioeg9S/Install Pro Tools 2019.12.0.pkg' matched from globbed '/private/tmp/dmg.Ioeg9S/Install*.pkg'.
Copier: Copied /private/tmp/dmg.Ioeg9S/Install Pro Tools 2019.12.0.pkg to /Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/ProTools2019-2019.12.0.pkg
{'Output': {}}
FileFinder
{'Input': {'find_method': 'glob',
           'pattern': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Codec '
                      'Installers/*.pkg'}}
FileFinder: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
FileFinder: Found file match: '/private/tmp/dmg.AlY1Z9/Codec Installers/Install Avid Codecs LE.pkg' from globbed '/private/tmp/dmg.AlY1Z9/Codec Installers/*.pkg'
FileFinder: DMG-relative file match: 'Codec Installers/Install Avid Codecs LE.pkg'
{'Output': {'dmg_found_filename': 'Codec Installers/Install Avid Codecs LE.pkg',
            'found_filename': '/private/tmp/dmg.AlY1Z9/Codec '
                              'Installers/Install Avid Codecs LE.pkg'}}
FlatPkgUnpacker
{'Input': {'destination_path': '/Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/unpack/codecs',
           'flat_pkg_path': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Codec '
                            'Installers/Install Avid Codecs LE.pkg'}}
FlatPkgUnpacker: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
FlatPkgUnpacker: Unpacked /private/tmp/dmg.nhqxvV/Codec Installers/Install Avid Codecs LE.pkg to /Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/unpack/codecs
{'Output': {}}
PackageInfoVersioner
{'Input': {'package_info_path': '/Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/unpack/codecs/AvidCodecsLE_component.pkg/PackageInfo'}}
PackageInfoVersioner: Found pkg_id com.avid.avidcodecsle.pkg
PackageInfoVersioner: Found version 2.7.3.39170
{'Output': {'pkg_id': 'com.avid.avidcodecsle.pkg', 'version': '2.7.3.39170'}}
FileFinder
{'Input': {'find_method': 'glob',
           'pattern': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Driver '
                      'Installers/*.pkg'}}
FileFinder: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
FileFinder: Found file match: '/private/tmp/dmg.zF0ata/Driver Installers/Install Avid HD Driver.pkg' from globbed '/private/tmp/dmg.zF0ata/Driver Installers/*.pkg'
FileFinder: DMG-relative file match: 'Driver Installers/Install Avid HD Driver.pkg'
{'Output': {'dmg_found_filename': 'Driver Installers/Install Avid HD '
                                  'Driver.pkg',
            'found_filename': '/private/tmp/dmg.zF0ata/Driver '
                              'Installers/Install Avid HD Driver.pkg'}}
FlatPkgUnpacker
{'Input': {'destination_path': '/Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/unpack/drivers',
           'flat_pkg_path': '/Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg/Driver '
                            'Installers/Install Avid HD Driver.pkg'}}
FlatPkgUnpacker: Mounted disk image /Volumes/EXT_DATA/Autopkgr/Manual_Imports/ProTools/Pro_Tools_2019.12.0_Mac.dmg
FlatPkgUnpacker: Unpacked /private/tmp/dmg.JEENQl/Driver Installers/Install Avid HD Driver.pkg to /Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/unpack/drivers
{'Output': {}}
PackageInfoVersioner
{'Input': {'package_info_path': '/Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/unpack/drivers/HD '
                                'Driver AppMan.pkg/PackageInfo'}}
PackageInfoVersioner: Found pkg_id com.avid.installer.osx.HDFamilyDriverAppMan
PackageInfoVersioner: Found version 19.12.0.534
{'Output': {'pkg_id': 'com.avid.installer.osx.HDFamilyDriverAppMan',
            'version': '19.12.0.534'}}
MunkiImporter
{'Input': {'MUNKI_REPO': '/Volumes/munki_repo',
           'additional_makepkginfo_options': ['--displayname',
                                              'Pro Tools 2019.12.0',
                                              '--pkgvers',
                                              '2019.12.0',
                                              '--requires',
                                              'ENTER CODECS MUNKI ITEM NAME '
                                              'HERE-2.7.3.39170',
                                              '--requires',
                                              'ENTER DRIVER MUNKI ITEM NAME '
                                              'HERE-19.12.0.534'],
           'pkg_path': '/Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/ProTools2019-2019.12.0.pkg',
           'pkginfo': {'catalogs': ['testing'],
                       'category': 'Music',
                       'description': 'The industry standard audio-production '
                                      'software for songwriters, musicians, '
                                      'producers, and engineers.',
                       'developer': 'Avid Technology, Inc.',
                       'maximum_os_version': '10.15.3',
                       'minimum_os_version': '10.13.6',
                       'name': 'ProTools2019',
                       'unattended_install': True},
           'repo_subdirectory': 'apps/protools'}}
MunkiImporter: Item ProTools2019-2019.12.0.pkg already exists in the munki repo as pkgs/apps/protools/ProTools2019-2019.12.0.pkg.
{'Output': {'pkg_repo_path': '/Volumes/munki_repo/pkgs/apps/protools/ProTools2019-2019.12.0.pkg'}}
PathDeleter
{'Input': {'path_list': ['/Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/unpack']}}
PathDeleter: Deleted /Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/unpack
{'Output': {}}
Receipt written to /Volumes/EXT_DATA/Autopkgr/Cache/com.github.apizz.munki.ProToolsImport/receipts/ProToolsTest.munki-receipt-20200629-164419.plist

Nothing downloaded, packaged or imported.
```

**Comment by apizz** [comment]
Confirmed resolved as part of #51
